rust/sbp/src/json/de.rs

```diff
@@ -44,12 +44,14 @@ pub fn stream_messages<R: futures::AsyncRead + Unpin>(
 #[derive(Debug, Default)]
 struct JsonDecoder {
     payload_buf: Vec<u8>,
+    eof_seen: bool,
 }
 
 impl JsonDecoder {
     fn new() -> Self {
         JsonDecoder {
             payload_buf: Vec::with_capacity(BUFLEN),
+            eof_seen: false,
```

Comment:
was trying to figure out a better way instead of holding this - seemed really hacky but i guess it works 👍 
- or maybe just `eprintln` inside decode, thought the dencode errors meant retry decoding so we needed two error types

Reply:
Agree that the solution is a bit hacky.  However, since these are library functions, it didn't feel right to add `eprintln` there (printing errors should be an application layer behavior).